### DISCUSSION TITLE
Python code runner inside neovim

### DISCUSSION THREAD

**Discussion by tonmoy998** [LazyVim/LazyVim#1271] [Ideas]
Kindly help me to run py file in neovim .I am trying to run autocmd FileType python and execude this code.

![Screenshot_20230801_201908](https://github.com/LazyVim/LazyVim/assets/74363531/de4c76c2-f383-4ddb-9c0f-0f3ba5c35a71)

**Comment by dpetka2001** [comment]
```lua
vim.keymap.set("n", "<leader>rp", "<cmd>TermExec cmd='python3 %' go_back=0<cr>", { desc = "Run Python file" })
```

**Reply by tonmoy998** [reply]
Thank you so much

**Reply by tonmoy998** [reply]
augroup code_exec
  autocmd!
autocmd FileType python nnoremap <buffer> <F12> :FloatermNew --autoclose=0 python3 % <CR> 
autocmd FileType javascript nnoremap <buffer> <F12> :FloatermNew --autoclose=0 nodejs % <CR> 
autocmd FileType bash,sh nnoremap <buffer> <F12> :FloatermNew --autoclose=0 bash % <CR> 
" autocmd FileType python nnoremap <buffer> <F12> :FloatermNew --autoclose=0 python3 % <CR> 
augroup END

Finally worked for me
